composer/trainer/trainer.py

```diff
@@ -2966,7 +2989,9 @@ def _eval_loop(
                 if dist_sampler is not None and drop_last == False and dataset_len is not None:
                     batch_num_samples_tensor = self.state.device.tensor_to_device(torch.tensor(rank_num_samples))
                     dist.all_reduce(batch_num_samples_tensor, reduce_operation='SUM')
-                    batch_num_samples = batch_num_samples_tensor.item()
+                    batch_num_samples = int(batch_num_samples_tensor.item())
+                    if batch_num_samples != batch_num_samples_tensor.item():
+                        raise ValueError('Number of samples in a batch should be an integer.')
```

Comment:
is this saying eval doesn't support SP?

Reply:
The `batch_num_samples` here is the sum of `batch_num_samples` across ranks in the sequence parallel group (see [this line](https://github.com/mosaicml/composer/blob/e6b334c908bf11e0935e252329ed577f5aa665e2/composer/trainer/trainer.py#L2991)). The check here only enforces that the sum is an integer.